R/proportional_hazards-glmnet.R

```diff
@@ -593,33 +603,39 @@ survival_prob_coxnet <- function(object,
     ...
   )
 
-  if (multi) {
+  if (length(penalty) > 1) {
     res_patched <- purrr::map(
       y,
       survfit_summary_to_patched_tibble,
       index_missing = missings_in_new_data,
       eval_time = eval_time,
       n_obs = n_obs
     )
+  } else {
+    res_patched <- survfit_summary_to_patched_tibble(
+      y,
+      index_missing = missings_in_new_data,
+      eval_time = eval_time,
+      n_obs = n_obs
+    )
+  }
+
+  if (multi) {
     res <- tibble::tibble(
       penalty = penalty,
       res_patched = res_patched
     ) %>%
-      tidyr::unnest(cols = res_patched) %>%
+      tidyr::unnest(cols = res_patched)
+    res_formatted <- res %>%
       keep_cols(output, keep_penalty = TRUE) %>%
       tidyr::nest(.pred = c(-.row)) %>%
       dplyr::select(-.row)
```

Comment:
This is such a minor thing. When reading this with fresh eyes I was confused why only the first part of `multi` created a `res` object. But it was an intermediate object between `res_patched` and `res_formatted`.

I wish it was one of the following ways instead

```r
  res_formatted <- tibble::tibble(
      penalty = penalty,
      res_patched = res_patched
    ) %>%
      tidyr::unnest(cols = res_patched)

    res_formatted <- res_formatted %>%
      keep_cols(output, keep_penalty = TRUE) %>%
      tidyr::nest(.pred = c(-.row)) %>%
      dplyr::select(-.row)
```

```r
    res_formatted <-  tibble::tibble(
      penalty = penalty,
      res_patched = res_patched
    ) %>%
      tidyr::unnest(cols = res_patched) %>%
      keep_cols(output, keep_penalty = TRUE) %>%
      tidyr::nest(.pred = c(-.row)) %>%
      dplyr::select(-.row)
```

Reply:
nono, good point! 👍 addressed in [1614ab6](https://github.com/tidymodels/censored/pull/279/commits/1614ab6119743e2fa0a5ae28c5e7b9e8a15c5c10)